### ISSUE TITLE
Issue with the G2P model

### ISSUE THREAD

**Issue by thelinhbkhn2014** [lingjzhu/CharsiuG2P#11]
Dear authors,
I have some questions regarding the use of your model. I used your model with the <eng-us> language, but it wrongly predicted many long words, even though these words appeared in the training set, such as "representation" and "multilingual." In my opinion, a model cannot be considered good if it incorrectly predicts words that are present in the training set. Could you please check your model?
Thanks!

**Comment by thelinhbkhn2014** [comment]
Example code:
from transformers import T5ForConditionalGeneration, AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained('google/byt5-small')
model = T5ForConditionalGeneration.from_pretrained('charsiu/g2p_multilingual_byT5_small_100')
words = ['<eng-uk>: representation']
out = tokenizer(words,padding=True,add_special_tokens=False,return_tensors='pt')
preds = model.generate(**out,num_beams=1)
phones = tokenizer.batch_decode(preds.tolist(),skip_special_tokens=True)

Output:  'ˌɹɛpɹəzɛnˈte'   
but the phoneme of "representation" in the training set: 'ˌɹɛpɹəzɛnˈteɪʃən'
This error also happens with many long words.

**Comment by lingjzhu** [comment]
Thank you for your interest. The output is too long in bytes so you need to set `max_length` to ensure that the output will not be truncated. For your case, the following code will work:
`words = ['<eng-US>: representation']
out = tokenizer(words,padding=True,add_special_tokens=False,return_tensors='pt')
preds = model.generate(**out,num_beams=1,max_length=50)
phones = tokenizer.batch_decode(preds.tolist(),skip_special_tokens=True)
phones`
Output: ˌɹɛpɹəzɛnˈteɪʃən
For `multilingual`, the output is `ˌməɫtiˈɫɪŋɡwəɫ`. 
Please note that the WER and PER of our model are not 100% and there will always be many errors. 
We have provided all the data and the training script. Do feel free to train your own model and help us improve it.

**Comment by thelinhbkhn2014** [comment]
Thank you for your response. 
Adjusting the max_length has helped, but I noticed that some long words, such as "multilingual" and "governmentally," are still not being predicted accurately. While I understand that your model's WER and PER are not perfect, I believe that words in the training set should be predicted correctly. Clearly, I can create a perfect pronunciation for these words by mapping them 1-1 from your data, instead of relying on your model :D 
Example: "multilingual" . Output: ˌməɫtiˈɫɪŋɡwəɫ  (Training set: ˌməɫtaɪˈɫɪŋwəɫ)
Anw, thank you for your contribution!

**Comment by lingjzhu** [comment]
Yeah most G2P systems would query the pronunciation dictionary first and only resort to a G2P model for out-of-vocabulary words. So your approach of mapping those words to a dictionary first is definitely the recommended approach.   

That said, your finding did help me find an error in the dictionary.  If you use Oxford or other dictionaries, the correct pronunciation of "multilingual" is [ˌməɫtiˈɫɪŋɡwəɫ] rather than what is in the eng-US dictionary (ˌməɫtaɪˈɫɪŋwəɫ or ˌməɫtiˈɫɪŋwəɫ). In the training set, words like monolingual or bilingual, their last syllables are all [ɡwəɫ].  In the eng-UK, multilingual is also pronounced as  [ɡwəɫ]. Maybe ˈɫɪŋwəɫ was treated as a noise by the model.

But yes, '<eng-uk>: governmentally' as input will produce the correct pronunciation, while '<eng-us>: governmentally' cannot. It's difficult to explain why.
